### PR TITLE
fix($window): allow $window to be mocked in unit tests

### PR DESCRIPTION
**Pull request by mattlewis92**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix

**What is the current behavior? (You can also link to an open issue here)**
https://github.com/angular/angular.js/issues/15685

**What is the new behavior (if this is a feature change)?**
N/A

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] <s>Docs have been added / updated (for bug fixes / features)</s>

**Other information**:

### CODE REVIEW

File: src/ng/jsonpCallbacks.js [hunk 1]
```diff
@@ -11,7 +11,7 @@
  */
 var $jsonpCallbacksProvider = /** @this */ function() {
   this.$get = ['$window', function($window) {
-    var callbacks = $window.angular.callbacks;
+    var callbacks = $window.angular ? $window.angular.callbacks : {};
```
Comment (gkalpak) [inline src/ng/jsonpCallbacks.js]: Actually, I think it makes more sense to just use `angular` directly:

```diff
-     var callbacks = $window.angular.callbacks;
+     var callbacks = angular.callbacks;
```

**Review by gkalpak — CHANGES_REQUESTED** [review]
Can you also change the commit message to `fix($jsonpCallbacks): ...`?

**Comment by mattlewis92** [comment]
@gkalpak done 😄

File: src/ng/jsonpCallbacks.js [hunk 2]
```diff
@@ -10,8 +10,8 @@
  * how they vary compared to the requested url.
  */
 var $jsonpCallbacksProvider = /** @this */ function() {
-  this.$get = ['$window', function($window) {
-    var callbacks = $window.angular.callbacks;
+  this.$get = [function() {
```
Comment (gkalpak) [inline src/ng/jsonpCallbacks.js]: We don't need the `[...]` now. It can be just `this.$get = function() {...`

Comment (mattlewis92) [inline src/ng/jsonpCallbacks.js]: Done

**Comment by mattlewis92** [comment]
CI failure is a flake - the API limit on saucelabs has been hit

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**